### PR TITLE
Gradle tweaks

### PR DESCRIPTION
**Pull request by snazy**
* configure-on-demand
* replace usages of deprecated Gradle APIs

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -51,18 +51,13 @@ subprojects {
...
     }
 
-    project(":polaris-service") {
-        apply plugin: "application"
-    }
-
-    project(":polaris-core") {
-        apply plugin: "java-library"
-    }
+    // Note: polaris-core seemed to be for Java 11, but uses functions that are introduced in later Java versions
```
Comment (RussellSpitzer) [inline build.gradle:59]: then shouldn't we just change the version? I'm confused why we would have this workaround?

**Comment by RussellSpitzer** [comment]
I really think we should break this one into separate PR's. The posfix -> . refactor I think is good but we are also doing a bunch of other things here.

**Comment by snazy** [comment]
> I really think we should break this one into separate PR's. The posfix -> . refactor I think is good but we are also doing a bunch of other things here.

That's all for the "replace deprecated Gradle API" part (`buildDir` is deprecated).
`sourceCompatibility`/`targetCompatibiltiy` didn't really work, see #76 (usage of newer Java APIs).

File: build.gradle [hunk 2]
```diff
@@ -51,17 +51,12 @@ subprojects {
...
     apply plugin: "groovy"
 
-    tasks.withType(JavaCompile) {
+    tasks.withType(JavaCompile).configureEach {
         options.compilerArgs << "-Xlint:unchecked"
         options.compilerArgs << "-Xlint:deprecation"
-    }
 
-    project(":polaris-service") {
-        apply plugin: "application"
```
Comment (RussellSpitzer) [inline build.gradle:60]: For my own reference, Plugin specification is moved into project specific build files.

**Review by RussellSpitzer — APPROVED** [review]

**Pull request merged**